sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/VariantAssignment.java

```diff
@@ -0,0 +1,231 @@
+import org.springframework.util.StringUtils;
+
+import com.azure.spring.cloud.feature.management.Variant;
+import com.azure.spring.cloud.feature.management.VariantProperties;
+import com.azure.spring.cloud.feature.management.implementation.models.Allocation;
+import com.azure.spring.cloud.feature.management.implementation.models.GroupAllocation;
+import com.azure.spring.cloud.feature.management.implementation.models.PercentileAllocation;
+import com.azure.spring.cloud.feature.management.implementation.models.UserAllocation;
+import com.azure.spring.cloud.feature.management.implementation.models.VariantReference;
+import com.azure.spring.cloud.feature.management.models.FeatureManagementException;
+import com.azure.spring.cloud.feature.management.models.TargetingException;
+import com.azure.spring.cloud.feature.management.targeting.TargetingContextAccessor;
+import com.azure.spring.cloud.feature.management.targeting.TargetingEvaluationOptions;
+import com.azure.spring.cloud.feature.management.targeting.TargetingFilterContext;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * Evaluator for Dynamic Feature and Feature Filters.
+ */
+public final class VariantAssignment {
+
+    private final TargetingContextAccessor contextAccessor;
+
+    private final TargetingEvaluationOptions evaluationOptions;
+
+    private final ObjectProvider<VariantProperties> propertiesProvider;
+
+    /**
+     * `Microsoft.TargetingFilter` evaluates a user/group/overall rollout of a feature.
+     * 
+     * @param contextAccessor Context for evaluating the users/groups.
+     * @param options enables customization of the filter.
+     */
+    public VariantAssignment(TargetingContextAccessor contextAccessor, TargetingEvaluationOptions options,
+        ObjectProvider<VariantProperties> propertiesProvider) {
+        this.contextAccessor = contextAccessor;
+        this.evaluationOptions = options;
+        this.propertiesProvider = propertiesProvider;
+    }
+
+    /**
+     * Assigns a Variant based on the allocations
+     * @param <T> type of the variant
+     * @param allocation Allocation percentage of the variants
+     * @param variants List of the possible variants.
+     * @return Variant object containing an instance of the type
+     */
+    public Mono<Variant> assignVariant(Allocation allocation, Collection<VariantReference> variants) {
+        TargetingFilterContext targetingContext = new TargetingFilterContext();
+
+        if (contextAccessor == null) {
+            throw new FeatureManagementException("No Targeting Filter Context found to assign variant.");
+        }
+
+        contextAccessor.configureTargetingContext(targetingContext);
+
+        String targetedUser = targetingContext.getUserId();
+
+        for (UserAllocation users : allocation.getUsers().values()) {
+            for (String user : users.getUsers().values()) {
+                if (targetedUser != null) {
```

Comment:
How about putting this if the outtest of this section? If a targedUser is null, we don't need to do the two for loops